### PR TITLE
initial component

### PR DESCRIPTION
**Pull request by iisa**
## `<ia-dropdown>` component
- uses any custom trigger
- draws list by a given option
- can draw a button or link inside dropdown
- on option selected: fires custom event `optionSelected` and also optional callback in an option body/model.
- has css vars to help toggle toggle

ezgif com-gif-maker (1)

### CODE REVIEW

File: demo/app-root.ts [hunk 1]
```diff
@@ -1,20 +1,98 @@
-import { html, css, LitElement } from 'lit';
-import { customElement } from 'lit/decorators.js';
-import '../src/your-webcomponent';
+/* eslint-disable import/no-duplicates */
+import { html, css, LitElement, SVGTemplateResult, svg } from 'lit';
+import { customElement, property } from 'lit/decorators.js';
+import '../src/ia-dropdown';
```
Comment (nsharma123) [inline demo/app-root.ts:4]: do we need to need to call `{ optionInterface }` explicitly in below line? I think you will get everything using this as well.

File: demo/app-root.ts [hunk 2]
```diff
@@ -1,20 +1,98 @@
...
...0 100" xmlns="http://www.w3.org/2000/svg"><g fill="none" fill-rule="evenodd"><path d="m93 95.6190775v4.3809225h-86l.00091489-4.3809225zm-3.3076923-8.0682881v5.8399885h-79.3846154v-5.8399885zm-2.469108-61.3914468.9542264.4869761.4775698 7.7869616.4775698 12.6529979v12.1688152l-.4775698 15.1688858-.0794428 10.2190486-1.3523534.4068998h-73.7729585l-1.4317962-.4068998-.5560994-10.2190486-.4784829-15.0878783v-12.1678841l.4784829-12.7349365.4364787-7.8288621.9953175-.4450756zm2.469108-11.0620805v8.7609138h-79.3846154v-8.7609138zm-39.6923077-15.0972621 43 9.75392865-1.7107266 3.00007055h-81.7325318l-2.5567416-2.4330181z" fill="#fff"/><path d="m51.6028793 29c7.1190869 7.5359754 22.8971207 15 15.9901161 35.2579416 0-13.8609831-3.3266359-16.0787404-11.6432258-24.3953303.228824 12.6744366.5667995 13.802523.1932361 24.374579-.1341927 3.8850801.3452129 9.4861753-2.6877233 12.6835908-3.8961215 4.1053124-11.3900578 6.002907-16.7795264 4.6437583-4.4827612-1.1344218-6.853801-6.228533-5.0902551-10.9486985 2.2450074-6.0280765 10.8278993-9.3863964 17.0615124-7.555321 1.2866177.3748446 2.711055 1.3636433 2.711055 1.3636433 0-12.4247052.4098317-23.8246554.244811-35.4241632z" fill="#2c2c2c"/></g></svg>`;
```
Comment (nsharma123) [inline demo/app-root.ts:18]: can we just move these svg icons in /assets directory and export from there?

File: src/ia-dropdown.ts [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+    }
+  }
+
+  toggleOptions() {
+    if (!this.open) {
+      this.open = true;
+      return;
+    }
+
+    this.open = false;
```
Comment (nsharma123) [inline src/ia-dropdown.ts:75]: You can make this bunch code in single liner fix.
`this.open = !this.open;`

File: src/ia-dropdown.ts [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+          ${this.options.map(o => this.renderOption(o))}
+        </ul>
+      </div>
+    `;
+  }
+
+  static styles = css`
+    :host {
+      display: inline;
+      font: sanserif;
```
Comment (nsharma123) [inline src/ia-dropdown.ts:129]: Is this font familty match with our parent font? why we need to define this font separately  ?

File: src/ia-dropdown.ts [hunk 5]
```diff
@@ -0,0 +1,286 @@
...
+    ul.dropdown-main.closed {
+      visibility: hidden;
+      height: 1px;
+      width: 1px;
+    }
+
+    ul.dropdown-main {
+      position: absolute;
+      list-style: none;
+      margin: 0px;
```
Comment (nsharma123) [inline src/ia-dropdown.ts:187]: let just use margin: 0 only and remove px.

File: demo/app-root.ts (hunk 1 above)
Comment (iisa) [reply demo/app-root.ts:4]: yah, it's how ts works.. take a look [here](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-8.html)

File: demo/app-root.ts [hunk 6]
```diff
@@ -1,20 +1,98 @@
...
+import { html, css, LitElement, SVGTemplateResult, svg } from 'lit';
+import { customElement, property } from 'lit/decorators.js';
+import '../src/ia-dropdown';
+import { optionInterface } from '../src/ia-dropdown';
 
 @customElement('app-root')
 export class AppRoot extends LitElement {
+  @property({ type: String, attribute: true, reflect: true }) colorScheme:
+    | 'light-bg'
+    | 'dark-bg' = 'dark-bg';
```
[the remark this answers is not in the document]
Comment (iisa) [reply demo/app-root.ts:12]: thanks for asking, i had an extra pipe in there.. actually, it should say:
`colorScheme: 'light-bg' | 'dark-bg' = 'dark-bg';`

`'light-bg' | 'dark-bg'` - this part of the code sets the only possible values of `colorScheme`.

File: demo/app-root.ts (hunk 2 above)
Comment (iisa) [reply demo/app-root.ts:18]: thanks for asking. this is demo so I will hold here.

File: src/ia-dropdown.ts (hunk 3 above)
Comment (iisa) [reply src/ia-dropdown.ts:75]: thanks!

File: src/ia-dropdown.ts [hunk 7]
```diff
@@ -0,0 +1,286 @@
...
+  get dropdownState(): string {
+    if (this.open) {
+      return 'open';
+    }
+
+    return 'closed';
+  }
+
+  get caretUp(): SVGTemplateResult {
+    return svg`<svg class="caret-up-svg" height="4" viewBox="0 0 8 4" width="8" xmlns="http://www.w3.org/2000/svg">
```
[the remark this answers is not in the document]
Comment (iisa) [reply src/ia-dropdown.ts:95]: thanks for the check here. the issue is that the `fill` needs to change color. if an image, it won't do that. will hold here.

File: src/ia-dropdown.ts (hunk 4 above)
Comment (iisa) [reply src/ia-dropdown.ts:129]: thank you for catching! removed

**Pull request merged**